float32[4] quad